gtt/gtt_display.py

```diff
@@ -182,3 +182,417 @@ def update_bar_value(self, bar_id: IdType, value: int):
+            if response == b'\xfc\x87\x00\x02\x00\x03':
+                print('Contact was made')
+
+    def create_toggle_region(self, region_id: IdType, x_pos: int, y_pos: int, width: int, height: int,
+                             off_id: int, on_id: int):
+        """Create a region of the screen that responds to touch events with a unique message and toggleable visual
+        update
+        :param region_id: index used to identify this touch region in the touch region list. Region 255 is reserved for
+         out of region responses
+        :param x_pos: the distance from the left edge of the screen in pixels
+        :param y_pos: the distance from the top edge of the screen in pixels
+        :param width: width of the trace region
+        :param height: height of the trace region
+        :param off_id: index of the loaded bitmap displayed when the region is in an inactive state
+        :param on_id: index of the loaded bitmap displayed when the region is in a toggled state
+        """
+
+        self._validate_x(x_pos, x_pos + width - 1)
+        self._validate_y(y_pos, y_pos + height - 1)
+        region_id = self._resolve_id(region_id, new=True)
+
+        self._conn.write(
+            bytes.fromhex('FE  96') +
+            region_id.to_bytes(1, 'big') +
+            ints_to_signed_shorts(x_pos, y_pos) +
+            ints_to_unsigned_shorts(width, height) +
+            off_id.to_bytes(1, 'big') +
+            on_id.to_bytes(1, 'big')
+        )
+
+        self._receive_status_response(252, 150)
+
+    def set_GPO_state(self, pin: int, state: bool):
+        """Toggle the specified General Purpose Output pin on or off, sourcing up to 15mA current at
+        5V per GPO or sinking to ground. This command can be used to control devices, or signal a host device.
+        :param pin: GPO to be controlled
+        :param state: GPO state, as per eGPOSetting Values
+        """
+
+        self._conn.write(
+            bytes.fromhex('FE  49') +
+            pin.to_bytes(1, 'big') +
+            state.to_bytes(1, 'big')
+        )
+
+    def setup_animation(self, animation_id: IdType, animation_instance: int, x_pos: int, y_pos: int, fileName: str,
+                        play=True):
+        """Define a region of the screen to be used for the specified animation. If an animation is already in use
+        at that index, it will be overwritten. Multiple Animation Instances can be setup from one buffered animation
+        file.
+        :param animation_id: index where an animation file has been loaded.
+        :param animation_instance: index used to identify this animation instance in the animation list.
+        :param x_pos: the distance from the left edge of the screen in pixels
+        :param y_pos: the distance from the top edge of the screen in pixels
+        :param fileName: filename, and path from the root folder, of the animation file to load.
+        :param play: desired animation state
```

Comment:
- lets change the first two parameters to `memory_id: IdType,  display_id: IdType`
- why are we specifying a `fileName`?
- The docstring could be clearer

Reply:
What is memory_id: IdType, display_id: IdType replacing?